be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -72,8 +72,59 @@ sealed class RollCallHandler(dbRef: => AskableActorRef) extends MessageHandler {
   }
 
   def handleCloseRollCall(rpcMessage: JsonRpcRequest): GraphMessage = {
+    // We first create the three (independent) futures
     val ask: Future[GraphMessage] = dbAskWritePropagate(rpcMessage)
-    Await.result(ask, duration) match {
+
+    // Creates a channel for each attendee (of name /root/lao_id/social/PublicKeyAttendee), returns a Future[GraphMessage]
+    def createAttendeeChannels(data: CloseRollCall, rpcMessage: JsonRpcRequest): Future[GraphMessage] = {
+      val listAttendeeChannels: List[(Channel, ObjectType.ObjectType)] = data.attendees.map(attendee => (generateSocialChannel(rpcMessage.getParamsChannel, attendee), ObjectType.CHIRP))
+      val askCreateChannels: Future[GraphMessage] = (dbActor ? DbActor.CreateChannelsFromList(listAttendeeChannels)).map{
+        case DbActorAck() => Left(rpcMessage)
+        case DbActorNAck(code, description) => Right(PipelineError(code, description, rpcMessage.id))
+        case _ => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, unknownAnswer, rpcMessage.id))
+      }
+      askCreateChannels
+    }
+
+    val askOldData = dbActor ? DbActor.ReadLaoData(rpcMessage.getParamsChannel)
+
+    rpcMessage.getParamsMessage match {
+      case Some(message: Message) =>
+        val data: CloseRollCall = message.decodedData.get.asInstanceOf[CloseRollCall]
+        val askCreateChannels: Future[GraphMessage] = createAttendeeChannels(data, rpcMessage)
+
+        // We create a new Future (tuple of Futures) to remove the obligatory sequential execution
+        // TODO: A possible addition would be to handle the error messages better
```

Comment:
how so? what's wrong with the current code?

Reply:
Nothing is actually wrong, however, if there are two different kinds of errors at the same time, we do not combine the two but just pick one

Reply:
all right thank you!